### DISCUSSION TITLE
Ergast data sanity check

### DISCUSSION THREAD

**Discussion by harningle** [theOehrly/Fast-F1#446] [General]
Following #432 and #433, I did a quick sanity check on Ergast database (their csv database image).  You can find my notebook here: https://gist.github.com/harningle/b6251082b6ad27451c268b3d6721865d.

I've checked:

* if the data types are correct according to their [user guide](http://ergast.com/docs/f1db_user_guide.txt)?
  - Yes all good
* can fastf1 handle the data correctly?
  - Yes. A bit surprisingly that problems in #432 are no longer there in Ergast. They seem to have fixed it already!
* are there any inconsistency within Ergast data
  - Yes. I **found one, and only one, inconsistency**. In the race results table (`results.csv`), the row `resultId = 19302` (Mike Hawthorn in 1954 British GP) has non-missing `time` (gap to the leader), but missing `milliseconds` (race time in milliseconds). If I understand it correctly, we should always be able to calculate `milliseconds` based on the race time of the winner plus the gap to the leader. So either both of them are missing, or both are non-missing

**Comment by theOehrly** [comment]
Nice work. Good to know that there are not really any other issues in the dataset. I suppose the best way to report this if it's only this one issue is here: http://ergast.com/mrd/bugs/

**Comment by harningle** [comment]
reported! Close this discussion for now
